File: Practice/longestSubstring/lengthOfLongestSubstring_test.go
package main

import "testing"

func TestLenghtOfLongestSubstring(t *testing.T) {
	input := "abcabcbb"
	output := lengthOfLongestSubstring(input)
	if output != 3 {
		t.Errorf("lengthOfLongestSubstring(%s) = %d; want 3", input, output)
	}

	input = "bbbbb"
	output = lengthOfLongestSubstring(input)
	if output != 1 {
		t.Errorf("lengthOfLongestSubstring(%s) = %d; want 1", input, output)
	}

	input = "pwwkew"
	output = lengthOfLongestSubstring(input)
	if output != 3 {
		t.Errorf("lengthOfLongestSubstring(%s) = %d; want 3", input, output)
	}
}

File: Practice/leetcode/problem_1_10/tests/length_of_longest_substring_test.rs
#[cfg(test)]
mod tests {
  use problem_1_10::length_of_longest_substring;

  #[test]
  fn test_length_of_longest_substring_1() {
    assert_eq!(
      length_of_longest_substring::length_of_longest_substring("".to_owned()),
      0
    );
  }

  #[test]
  fn test_length_of_longest_substring_2() {
    assert_eq!(
      length_of_longest_substring::length_of_longest_substring("abcabcbb".to_owned()),
      3
    );
  }

  #[test]
  fn test_length_of_longest_substring_3() {
    assert_eq!(
      length_of_longest_substring::length_of_longest_substring("bbbbb".to_owned()),
      1
    );
  }

  #[test]
  fn test_length_of_longest_substring_4() {
    assert_eq!(
      length_of_longest_substring::length_of_longest_substring("pwwkew".to_owned()),
      3
    );
  }

  #[test]
  fn test_length_of_longest_substring_5() {
    assert_eq!(
      length_of_longest_substring::length_of_longest_substring("pqrstuv".to_owned()),
      7
    );
  }
}
